src/Components/Popup/Popup.tsx

```diff
@@ -1,24 +1,30 @@
 import * as React from "react";
 import * as PropTypes from "prop-types";
 
+import { TriggerEvents } from "../ExpandControl";
 import { ExpandContext, ExpandContextTypes } from "../ExpandController";
-import { PopupProps, PopupPropTypes } from "./PopupProps";
+import { PopupDefaultProps, PopupProps, PopupPropTypes } from "./PopupProps";
 
 export class Popup extends React.Component<PopupProps> {
     public static readonly propTypes = PopupPropTypes;
     public static readonly contextTypes = ExpandContextTypes;
+    public static readonly defaultProps = PopupDefaultProps;
 
     public readonly context: ExpandContext;
 
     public render() {
-        const { popupId, ...childProps } = this.props;
+        const { popupId, triggerEvent, ...childProps } = this.props;
 
         if (!this.context.isExpanded(popupId)) {
             return null;
         }
 
+        Object.assign(childProps, {
+            [triggerEvent]: this.context.changeExpandState(popupId, false)
```

Comment:
popup element is controlling by expandControl. Why it changingExpand state by itself? 
If we have expandControl, we does not needed popup component. Any HTML element with data attribute can be popup

Reply:
So how will we control hiding popup on expand state equaled to false?